src/thor/triplegbuilder.cc

```diff
@@ -481,6 +481,43 @@ void AddTransitNodes(TripLeg_Node* trip_node,
   }
 }
 
+void SetTripEdgeRoadClass(TripLeg_Edge* trip_edge,
+                          const DirectedEdge* directededge,
+                          const GraphTile* graphtile,
+                          GraphReader& graphreader) {
+  trip_edge->set_road_class(GetTripLegRoadClass(directededge->classification()));
+  // If this is a ramp it may have been reclassified in graph enhancer.
+  // To restore the original road class for motorway_links, we check if any of the adjacent edges is
+  // a motorway.
+  if (directededge->use() == Use::kRamp) {
+    const DirectedEdge* opposing_edge = graphreader.GetOpposingEdge(directededge, graphtile);
+    for (const auto& edge : {directededge, opposing_edge}) {
+      if (!edge || !graphreader.GetGraphTile(edge->endnode(), graphtile)) {
+        // If this edge was invalid or we couldn't get the opposing edge's tile, skip
+        continue;
+      }
+      // check edges leaving node
+      for (const auto& edge : graphtile->GetDirectedEdges(edge->endnode())) {
```

Comment:
you are shadowing the previously defined `edge` here, maybe name the previous one `directed_edge` or something

Reply:
Good catch. Wonder if there's a lint rule we can enable for variable shadowing.

Reply:
Actually the compiler caught it
```
/root/project/src/thor/triplegbuilder.cc:500:59: error: use of 'edge' before deduction of 'auto'
       for (const auto& edge : graphtile->GetDirectedEdges(edge->endnode())) {
                                                           ^```

Reply:
I was about to say the same (compiler should be warning about shadowing)